src/figuresClasses.ts

```diff
@@ -1,19 +1,101 @@
-export interface Figure {
+type Colors = 'red' | 'green' | 'blue';
 
+enum Shapes {
+  triangle = 'triangle',
+  circle = 'circle',
+  rectangle = 'rectangle',
+}
+export interface Figure {
+  color: Colors;
+  shape: Shapes;
+  getArea(): number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape: Shapes;
 
-}
+  getArea(): number {
+    const semiPerimeter = (this.sideA + this.sideB + this.sideC) / 2;
+
+    return +Math.sqrt(semiPerimeter
+      * (semiPerimeter - this.sideA)
+      * (semiPerimeter - this.sideB)
+      * (semiPerimeter - this.sideC)).toFixed(2);
+  }
+
+  private hasInvalidShape(): boolean {
+    const sides = [this.sideA, this.sideB, this.sideC];
+
+    const maxSide = sides.sort((a, b) => a - b).pop();
+    const twoSideSum = sides.reduce((sum, current) => sum + current);
+    const hasInvalidValue = sides.some((side) => side <= 0);
+
+    if (!maxSide) {
+      return false;
+    }
 
-export class Circle {
+    return (maxSide >= twoSideSum) || hasInvalidValue;
+  }
 
+  constructor(
+    public color: Colors,
+    public sideA: number,
+    public sideB: number,
+    public sideC: number,
+  ) {
+    this.shape = Shapes.triangle;
+
+    if (this.hasInvalidShape()) {
+      throw new Error(
+        'It`s not possible to create a triangle with these parameters',
```

Comment:
You could print out the params in the error

Reply:
+